### PR TITLE
[Hack Update] Refresh of 001-IntroToKubernetes

### PR DESCRIPTION
**Pull request by jrzyshr**
This PR is a "refresh" to the original What The Hack: Intro to Kubernetes!  The hack has been updated to use the latest What The Hack templates as per the [WTH Contribution Guide](https://aka.ms/wthcontribute), as well as minor updates to the technical content.

Updates include:
	• Move everything to new templates
	• Update Student Guide to refer to "Resources.zip" file. 
	• Update Coach Guide to refer to "Resources.zip" file and how coaches need to create/distribute it!
	• Add diagram to coach guide explaining the optional paths for C1/C2/C3. Paths A, B, C, D.
	• Fixed up student guide for C1 to spell out details of how to deploy ARM template for Linux VM as it is not core to the hack.
	• Verified all references to Student resource files use the new padded #s. (Challenge-XX) folders.
	• Solution 2B - updated coach guide to refer to the Resource.zip C2 folder for sample Dockerfiles!
	• Checked all Solution files for references to /Coach/Solutions folders with new Challenge-XX folder name!
	• Put Lecture.pptx on a diet - Stripped out all old redundant themes and made it consistent.
	• Update FabMedical app code & containers to latest version from MCW source?  Thanks Larry Claman!
- Rebuild app and deployed updated containers to Dockerhub

### CODE REVIEW

File: 001-IntroToKubernetes/Coach/Solution-01A.md [hunk 1]
```diff
@@ -56,7 +59,7 @@ One of the tasks in this challenge is, prior to building the docker images, is t
 	- **NOTE:** Attendees should **not** struggle with getting app to run locally.  If they are not familiar with Node.js, help them through this part.  Let them spend more time figuring out Docker later.
```
Comment (perktime) [inline 001-IntroToKubernetes/Coach/Solution-01A.md:59]: add "the" between "getting" and "app"

File: 001-IntroToKubernetes/Coach/Solution-02.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+ Use this path to provide students with a comprehensive experience of building, testing, and running containers locally. Students will create a VM (using a provided ARM template) with Docker and all other required tools to ensure uniformity of environments. In this path, your students will create a Dockerfile, build and test local containers, and then push these container images to Azure Container Registry.
+
+#### [PATH B](Solution-02B.md) - Challenge 2B Coach's Guide
+
+Use this path if your students understand Docker, don't care about building images locally, and/or have environments issues that would prevent them from building containers locally. In this path, your students will be given a Dockerfile, will create an Azure Container Registry, and then will use ACR tasks to build the images natively inside ACR.
+
+#### [PATH C](Solution-02C.md) - Challenge 2C Coach's Guide
+
+Use this path to skip any work with docker & containers, but you still want to learn about ACR.  The path is similar to Path B, however in this case students will simply import pre-staged containers from DockerHub into ACR (eg, there is no container building involved)
```
Comment (perktime) [inline 001-IntroToKubernetes/Coach/Solution-02.md:25]: Capitalize docker

File: 001-IntroToKubernetes/Coach/Solution-02B.md [hunk 3]
```diff
@@ -1,12 +1,13 @@
-# Challenge 1b & 2b: Coach's Guide
+# Challenge 02 - Path B: The Azure Container Registry - Coach's Guide 
 
-[< Previous Challenge](01-containers.md) - **[Home](README.md)** - [Next Challenge >](03-k8sintro.md)
+[< Previous Solution](./Solution-01.md) - **[Home](./README.md)** - [Next Solution >](./Solution-03.md)
 
 ## Challenge 1 & 2 Path B
 
 This is **PATH B**: Use this path if your students understand docker, don't care about building images locally, and/or have environments issues that would prevent them from building containers locally. In this path, your students will be given a Dockerfile, will create an Azure Container Registry, and then will use ACR tasks to build the images natively inside ACR.
```
Comment (perktime) [inline 001-IntroToKubernetes/Coach/Solution-02B.md:7]: Capitalize docker

File: 001-IntroToKubernetes/Student/Challenge-01A.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+[< Previous Challenge](./Challenge-01.md) - **[Home](../README.md)** - [Next Challenge >](./Challenge-02A.md)
+
+## Introduction
+
+The first step in our journey will be to take our application and package it as a container image using Docker.
+
+## Description
+
+To help ensure everyone can start this hack on the same page and avoid the complexities of installing and setting up Docker, we have provided you with an ARM Template that will deploy a Linux VM into Azure. Once deployed, the Linux VM will have the FabMedical application's source code and all of the Docker & Node JS tools pre-installed.
```
Comment (perktime) [inline 001-IntroToKubernetes/Student/Challenge-01A.md:11]: The correct spelling of Node JS is Node.js. Might want to change throughout this section

File: 001-IntroToKubernetes/Student/Challenge-08.md [hunk 5]
```diff
@@ -77,10 +77,10 @@ In this challenge we will be creating Azure data disks and using the Kubernetes
 
 ## Success Criteria
 
-1. Speaker and session data is imported into Mongo
-1. Stop and restart the MongoDB pod and make sure the data wasn't lost.
+1. Verify that speaker and session data is imported into Mongo
```
Comment (perktime) [inline 001-IntroToKubernetes/Student/Challenge-08.md:80]: Change Mongo to MongoDB

File: 001-IntroToKubernetes/Student/Challenge-11.md [hunk 6]
```diff
@@ -30,16 +30,16 @@ In this challenge you will learn how to view application logs and trouble-shoot
 ## Success Criteria
 
 1. Show logs for the containers running in your cluster.
-2. Log into a running container and issue bash commands.
+2. Demonstrate that you can log into a running container and issue bash commands.
```
Comment (perktime) [inline 001-IntroToKubernetes/Student/Challenge-11.md:33]: Change "log into" to "login to"

**Review by perktime — CHANGES_REQUESTED** [review]

File: 001-IntroToKubernetes/Coach/README.md [hunk 7]
```diff
@@ -1,31 +1,97 @@
...
+### Pre-Select Your Path For Container Content
+Coaches, be sure to read the [Coach Guidance for Challenge 1](./Solution-01.md). You will need to select a proper path based on the learning objectives of the organization (to be decided PRIOR TO the hack!).  Select the proper path after consulting with the organization's stakeholder(s) for the hack.
+
+## Azure Requirements
+
+This hack requires students to have access to an Azure subscription where they can create and consume Azure resources. These Azure requirements should be shared with a stakeholder in the organization that will be providing the Azure subscription(s) that will be used by the students.
+
+- Attendees should have the “Azure account administrator” (or "Owner") role on the Azure subscription in order to authenticate their AKS clusters against their Azure Container Registries.  For more info: <https://docs.microsoft.com/en-us/azure/aks/cluster-container-registry-integration>
+- Each student will spin up the following resources in Azure:
+	- 3 x 2core VMs for the AKS cluster + 2 Public IPs (1 for AKS cluster, 1 for content-web)
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/README.md:61]: Done.

File: 001-IntroToKubernetes/Coach/README.md [hunk 8]
```diff
@@ -1,31 +1,97 @@
...
+	- 1 x 2core VM for the Docker Build machine + one Public IP
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/README.md:62]: Done.

File: 001-IntroToKubernetes/Coach/README.md [hunk 9]
```diff
@@ -1,31 +1,97 @@
...
+	- Total: 8 cores + 3 Public IPs per attendee
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/README.md:63]: Done.

File: 001-IntroToKubernetes/Coach/README.md [hunk 10]
```diff
@@ -1,31 +1,97 @@
...
+
+- `./Coach`
+  - Coach's Guide and related files
+  - [Lecture presentation](Coach/Lectures.pptx) with short presentations to introduce each challenge.
+- `./Coach/Solutions`
+  - Solution files with completed example answers to a challenge
+- `./Student`
+  - Student's Challenge Guide
+- `./Student/Resources`
+  - Resource files, sample code, scripts, etc meant to be provided to students. (Must be packaged up by the coach and provided to students at start of event)
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/README.md:97]: Done.

File: 001-IntroToKubernetes/Coach/Solution-00.md [hunk 11]
```diff
@@ -17,10 +17,12 @@ If attendees will be using a shared Azure subscription, you should be aware of t
 	- 3 x 2core VMs for the AKS cluster + 2 PIPs. 1 for AKS cluster, 1 for content-web
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-00.md:17]: Done.

File: 001-IntroToKubernetes/Coach/Solution-01.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+ Use this path to provide students with a comprehensive experience of building, testing, and running containers locally. Students will create a VM (using a provided ARM template) with Docker and all other required tools to ensure uniformity of environments. In this path, your students will create a Dockerfile, build and test local containers, and then push these container images to Azure Container Registry. 
+#### [PATH B](./Solution-02B.md) - Challenge 2B Coach's Guide
+Use this path if your students understand Docker, don't care about building images locally, and/or have environments issues that would prevent them from building containers locally. In this path, your students will be given a Dockerfile, will create an Azure Container Registry, and then will use ACR tasks to build the images natively inside ACR.
+#### [PATH C](./Solution-02C.md) - Challenge 2C Coach's Guide
+Use this path to skip any work with docker & containers, but you still want to learn about ACR.  The path is similar to Path B, however in this case students will simply import pre-staged containers from DockerHub into ACR (eg, there is no container building involved)
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-01.md:33]: Done.

File: 001-IntroToKubernetes/Coach/Solution-01A.md [hunk 13]
```diff
@@ -1,12 +1,15 @@
-# Challenge 1a: Coach's Guide
+# Challenge 01 - Path A: Got Containers? - Coach's Guide 
 
-[< Previous Challenge](./00-prereqs.md) - **[Home](README.md)** - [Next Challenge >](./02a-acr.md)
+[< Previous Solution](./Solution-01.md) - **[Home](./README.md)** - [Next Solution >](./Solution-02A.md)
 
 ## Challenge 1, Path A
 
 This is **PATH A**: Use this path if your students want to understand what's involved in creating a docker container, and understand basic docker commands.  In this path, your students will create a Dockerfile, build and test local containers, and then push these container images to Azure Container Registry.
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-01A.md:7]: Done.  FYI... from the Docker website, it should be capitalized unless referring to the actual CLI commands.
https://www.docker.com/legal/trademark-guidelines/

File: 001-IntroToKubernetes/Coach/Solution-01A.md [hunk 14]
```diff
@@ -1,12 +1,15 @@
...
 
-### Coach Notes:
+## Notes & Guidance
+
+**NOTE:** Before the hack, it is the Coach's responsibility to download and package up the contents of the `/Student/Resources` folder of this hack into a `Resources.zip` file. The coach should then provide a copy of the `Resources.zip` file to all students at the start of the hack.  The student guide refers to relative path locations within this zip package.
+
 #### Tooling:
 The students can either work with Docker locally on their workstation, or they can deploy a build machine VM in Azure using the provided template.  There is no need to do both.  We recommend using the build VM deployed in Azure as this provides a known-to-work common environment.
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-01A.md:14]: Done.

File: 001-IntroToKubernetes/Coach/Solution-01A.md (hunk 1 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-01A.md:59]: Done!

File: 001-IntroToKubernetes/Coach/Solution-02.md (hunk 2 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-02.md:25]: Done.

File: 001-IntroToKubernetes/Coach/Solution-02B.md (hunk 3 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-02B.md:7]: Done.

File: 001-IntroToKubernetes/Student/Challenge-01A.md (hunk 4 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Student/Challenge-01A.md:11]: Done. (This probably appears in more locations than this.  Really old content!)

File: 001-IntroToKubernetes/Student/Challenge-08.md (hunk 5 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Student/Challenge-08.md:80]: Done.

File: 001-IntroToKubernetes/Student/Challenge-11.md (hunk 6 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Student/Challenge-11.md:33]: Done.

File: 001-IntroToKubernetes/Coach/README.md [hunk 15]
```diff
@@ -1,31 +1,97 @@
...
-   - Explore the logs provided by Kubernetes using the Kubernetes CLI, configure Azure Monitor and build a dashboard that monitors your AKS cluster
\ No newline at end of file
+
+- Challenge 00: **[Prerequisites - Ready, Set, GO!](./Solution-00.md)**
+	 - Prepare your workstation to work with Azure, Docker containers, and AKS.
+- Challenge 01: **[Got Containers?](./Solution-01.md)**
+	 - Package the "FabMedical" app into a Docker container and run it locally.
+- Challenge 02: **[The Azure Container Registry](./Solution-02.md)**
+	 - Deploy an Azure Container Registry, secure it and publish your container.
+- Challenge 03: **[Introduction To Kubernetes](./Solution-03.md)**
```
Comment (Whowong) [inline 001-IntroToKubernetes/Coach/README.md:19]: Minor change to standardize both titles.  The solution file says "Intro To Kubernetes"

File: 001-IntroToKubernetes/Coach/Solution-11.md [hunk 16]
```diff
@@ -1,6 +1,6 @@
-# Challenge 11: Coach's Guide
+# Challenge 11 - Operations and Monitoring - Coach's Guide 
 
-[< Previous Challenge](./10-networking.md) - **[Home](README.md)**
+[< Previous Solution](./Solution-10.md) - **[Home](./README.md)** - [Next Solution >](./Solution-12.md)
```
Comment (Whowong) [inline 001-IntroToKubernetes/Coach/Solution-11.md:3]: Remove solution 12 as there are only 11 challenges

File: 001-IntroToKubernetes/README.md [hunk 17]
```diff
@@ -1,44 +1,46 @@
...
+	 - Flex Kubernetes' muscles by scaling pods, and then nodes. Observe how Kubernetes responds to resource limits.
+- Challenge 06: **[Deploy MongoDB to AKS](Student/Challenge-06.md)**
+	 - Deploy MongoDB to AKS from a public container registry.
+- Challenge 07: **[Updates and Rollbacks](Student/Challenge-07.md)**
+	 - Deploy v2 of FabMedical to AKS via rolling updates, roll it back, then deploy it again using the blue/green deployment methodology.
+- Challenge 08: **[Storage](Student/Challenge-08.md)**
+	 - Delete the MongoDB you created earlier and observe what happens when you don't have persistent storage. Fix it!
+- Challenge 09: **[Helm](Student/Challenge-09.md)**
+	 - Install Helm tools, customize a sample Helm package to deploy FabMedical, publish the Helm package to Azure Container Registry and use the Helm package to redeploy FabMedical to AKS.
+- Challenge 10: **[Networking](Student/Challenge-10.md)**
```
Comment (Whowong) [inline 001-IntroToKubernetes/README.md:39]: Need to confirm title as Challenge 10 in the markdown file shows "Networking, DNS, and Ingress"

**Review by Whowong — CHANGES_REQUESTED** [review]

File: 001-IntroToKubernetes/Coach/README.md (hunk 15 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/README.md:19]: Done.  Changed S3 to "Introduction"

File: 001-IntroToKubernetes/Coach/Solution-11.md (hunk 16 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/Coach/Solution-11.md]: Done.

File: 001-IntroToKubernetes/README.md (hunk 17 above)
Comment (jrzyshr) [reply 001-IntroToKubernetes/README.md:39]: Done.  To make it consistent, I changed it to "Networking and Ingress" everywhere.  We do plan to change/update this challenge in a future PR

**Review by perktime — APPROVED** [review]

**Pull request merged**